tfutils/base.py

```diff
@@ -271,17 +271,38 @@ def initialize(self):
                 self.load_rec()
             if self.load_data is not None:
                 rec, cache_filename = self.load_data
+                # get variables to restore
+                restore_vars = self.get_restore_vars(cache_filename)
+                log.info('Restored Vars:\n'+str([restore_var.name for restore_var in restore_vars]))
+                tf_saver_restore = tf.train.Saver(restore_vars)
                 # tensorflow restore
                 log.info('Restoring variables from record %s (step %d)...' % (str(rec['_id']), rec['step']))
-                tf_saver.restore(self.sess, cache_filename)
+                tf_saver_restore.restore(self.sess, cache_filename)
                 log.info('... done restoring.')
+                self.sess.run(tf.variables_initializer(list(var for var in tf.global_variables()+tf.local_variables() \
```

Comment:
@ewang314 this is pretty nasty line of code.   can you simply it so it's more readable?